acceptance/localcluster/docker.go

```diff
@@ -0,0 +1,295 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Peter Mattis (peter.mattis@gmail.com)
+
+package localcluster
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strconv"
+
+	"github.com/samalba/dockerclient"
+)
+
+func getTLSConfig() *tls.Config {
+	certPath := os.Getenv("DOCKER_CERT_PATH")
+
+	clientCert, err := tls.LoadX509KeyPair(
+		filepath.Join(certPath, "cert.pem"),
+		filepath.Join(certPath, "key.pem"),
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	rootCAs := x509.NewCertPool()
+	caCert, err := ioutil.ReadFile(filepath.Join(certPath, "ca.pem"))
+	if err != nil {
+		panic(err)
+	}
+	rootCAs.AppendCertsFromPEM(caCert)
+
+	return &tls.Config{
+		Certificates: []tls.Certificate{clientCert},
+		RootCAs:      rootCAs,
+	}
+}
+
+// newDockerClient constructs a new docker client using the best available
+// method. On darwin (Mac OS X), first look for the DOCKER_HOST env
+// variable. If not found, run "boot2docker shellinit" and add the exported
+// variables to the environment. If DOCKER_HOST is set, initialize the client
+// using DOCKER_TLS_VERIFY and DOCKER_CERT_PATH. If DOCKER_HOST is not set,
+// look for the unix domain socket in /run/docker.sock and
+// /var/run/docker.sock.
+func newDockerClient() dockerclient.Client {
+	if runtime.GOOS == "darwin" && os.Getenv("DOCKER_HOST") == "" {
+		output, err := exec.Command("boot2docker", "shellinit").Output()
```

Comment:
This is the wrong place to help out: if you don't have `DOCKER_HOST` set then builder.sh will fail (and be ignored because run.sh doesn't `set -e`). If we're going to take responsibility for `shellinit` then it needs to happen in `run.sh` (or we need to fail with a helpful message).